### PR TITLE
Institution configuration options API

### PR DESCRIPTION
**Pull request by arothuis**
This adds the ability to query Middleware for InstitutionConfigurationOptions.

### CODE REVIEW

File: src/Surfnet/StepupMiddlewareClientBundle/Configuration/Service/InstitutionConfigurationOptionsService.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+    public function getInstitutionConfigurationOptionsFor($institution)
+    {
+        $data = $this->service->getInstitutionConfigurationOptionsFor($institution);
+
+        if ($data === null) {
+            return null;
+        }
+
+        $institutionConfigurationOptions = InstitutionConfigurationOptions::fromData($data[$institution]);
```
Comment (arothuis) [inline src/Surfnet/StepupMiddlewareClientBundle/Configuration/Service/InstitutionConfigurationOptionsService.php]: I think trying to get the options for the institution out of the response data is kind of weird, but it's the result of relying on the JsonSerialized data on the Middleware side. We could modify the Middleware to return only the options, without the Institution as a key.

If the institution were to remain a key, some checking has to be done before assuming it is present in the `$data` array.

Thoughts, @DRvanR?

File: src/Surfnet/StepupMiddlewareClientBundle/Tests/Configuration/Service/InstitutionConfigurationOptionsServiceTest.php [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+use Surfnet\StepupMiddlewareClientBundle\Exception\InvalidResponseException;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class InstitutionConfigurationOptionsServiceTest extends TestCase
+{
+    /**
+     * @group institution-configuration
+     */
+    public function testQueriedInstitutionConfigurationOptionsAreConvertedToADtoCorrectly()
```
Comment (arothuis) [inline src/Surfnet/StepupMiddlewareClientBundle/Tests/Configuration/Service/InstitutionConfigurationOptionsServiceTest.php]: Maybe add tech debt to unify (test) code style for all SSA projects?

**Comment by arothuis** [comment]
@DRvanR

**Comment by arothuis** [comment]
That's weird. I rebased my commits to be in the correct order, but nothing changed in this branch before pushing. Checking this out.

**Comment by DRvanR** [comment]
This should be adapted to match the changes made in  https://github.com/SURFnet/Stepup-Middleware/pull/148

**Comment by arothuis** [comment]
@DRvanR

File: src/Surfnet/StepupMiddlewareClientBundle/Tests/Configuration/Service/InstitutionConfigurationOptionsServiceTest.php (hunk 2 above)
Comment (DRvanR) [inline src/Surfnet/StepupMiddlewareClientBundle/Tests/Configuration/Service/InstitutionConfigurationOptionsServiceTest.php]: Good idea!

**Comment by DRvanR** [comment]
:+1:

**Pull request merged**